src/mono/wasm/runtime/memory.ts

```diff
@@ -83,25 +89,30 @@ export function setU32_unchecked(offset: MemOffset, value: NumberOrPointer): voi
 
 export function setU32(offset: MemOffset, value: NumberOrPointer): void {
     assert_int_in_range(<any>value, 0, 0xFFFF_FFFF);
+    updateGrowableHeapViews();
     Module.HEAPU32[<any>offset >>> 2] = <number><any>value;
 }
 
 export function setI8(offset: MemOffset, value: number): void {
     assert_int_in_range(value, -0x80, 0x7F);
+    updateGrowableHeapViews();
     Module.HEAP8[<any>offset] = value;
 }
 
 export function setI16(offset: MemOffset, value: number): void {
     assert_int_in_range(value, -0x8000, 0x7FFF);
+    updateGrowableHeapViews();
     Module.HEAP16[<any>offset >>> 1] = value;
 }
 
 export function setI32_unchecked(offset: MemOffset, value: number): void {
+    updateGrowableHeapViews();
```

Comment:
Have you been able to benchmark the impact of this change? I'm concerned it will be a measurable performance regression in single-threaded builds for no upside

Reply:
`updateGrowableHeapViews` is NOOP in single threaded, because rollup. I will double check that .js file to make sure.

Reply:
This is MT version
![image](https://github.com/dotnet/runtime/assets/271576/ad12876d-f8b4-461e-9c04-6e47dc19e544)

This is single threaded version
![image](https://github.com/dotnet/runtime/assets/271576/a8468300-9eb9-4985-96e8-e32590230053)

The string `"Module.wasmMemory not set"` is not even in the file.

I will remove the `mono_assert` for MT performance sake.